subsys/bluetooth/host/gatt.c

```diff
@@ -2308,6 +2325,17 @@ static int gatt_indicate(struct bt_conn *conn, uint16_t handle,
 		return -EPERM;
 	}
 
+	/* Check if client has subscribed before sending indications, except in
+	 * the case of a service changed indication.
+	 * This is not really required in the Bluetooth specification, but
+	 * follows its spirit.
+	 */
+	if (!(bt_gatt_is_subscribed(conn, params->attr, BT_GATT_CCC_INDICATE) ||
+	      is_sc_ind)) {
```

Comment:
I wonder if this is more logical:
```suggestion
	if (!bt_gatt_is_subscribed(conn, params->attr, BT_GATT_CCC_INDICATE) && !is_sc_ind)) {
```

Which gives the same result, but I think it's more clear what we really are checking here.

Reply:
yep that's clearer, thanks